Re-drafted acceptance criteria (#40)
form3tech-oss/interview-accountapi: README.md

# Form3 Take Home Exercise

> Senior Engineers at Form3 build highly available distributed systems in a microservices environment. Our take home test is designed to evaluate real world activities that are involved with this role. We recognise that this may not be as mentally challenging and may take longer to implement than some algorithmic tests that are often seen in interview exercises. Our approach however helps ensure that you will be working with a team of engineers with the necessary practical skills for the role (as well as a diverse range of technical wizardry). 

Reviewer: I was thinking an intro might be useful to try to justify why we're not asking for a red-black tree optimisation algorithm or some other simple short exercise. Was hoping we could find some words that would make the exercise & form3 more compelling. Not sure I've done that here though!

Maybe it should have some links to more info about how we work, e.g. the .tech podcasts.

If you encounter any problems running the fake account API we would encourage you to do some debugging first,
before reaching out for help.

> ## Acceptance Critera

Reviewer: I think this title is opening us up to lots of questions when we reject poor solutions. I preferred the original headings.

### Shoulds
The finished solution **should:**
- Be written in Go.
- Be a client library suitable for use in another software project.
- Implement the `Create`, `Fetch`, and `Delete` operations on the `accounts` resource.
- Be well tested to the level you would expect in a commercial environment.
- Contain documentation of your technical decisions.
- Be simple and concise.
- Have tests that run from `docker-compose up` - our reviewers will run `docker-compose up` to assess if your tests pass.

### Should Nots
The finished solution **should not:**
- Use a code generator to write the client library.
- Use (copy or otherwise) code from any third party without attribution to complete the exercise, as this will result in the test being rejected.
- Use a library for your client (e.g: go-resty). Only test or non-client libraries are allowed (e.g. for UUID types).
- Implement an authentication scheme.
- Implement support for the fields `data.attributes.private_identification`, `data.attributes.organisation_identification`
  and `data.relationships`, as they are omitted in the provided fake account API implementation.

> - Have advanced features, like retries, metrics or tracing.

Reviewer: ```suggestion
- Have advanced features, however discussion of anything extra you'd expect a production client to contain would be useful in the documentation. 
```

- Be a command line client or other type of program - the requirement is to write a client library.
- Implement the `List` operation.

> 

Reviewer: Should we say anything about validation? I've reviewed quite a few tests where they have clearly spent days on client side validation. It's useful to know what candidates think is necessary. Not sure how to write something in the file without being too prescriptive though, e.g. `- if you choose to add validation, demonstrate only a sample of fields`

If you encounter any problems running the fake account API we would encourage you to do some debugging first,
before reaching out for help.

## Acceptance Critera
### Shoulds
The finished solution **should:**
- Be written in Go.

> - Be a client library suitable for use in another software project.

Reviewer: Comments from talent team that feedback suggests we don't clearly ask for production ready code (only commercially tested)
```suggestion
- Be a client library suitable for use within a production microservice.  
```
